model-scoring-core/src/main/scala/org/trustedanalytics/scoring/ScoringService.scala

```diff
@@ -129,37 +138,103 @@ class ScoringService(model: Model) extends Directives {
         requestUri { uri =>
           get {
             import spray.json._
-            onComplete(Future { model.modelMetadata() }) {
+            require(scoringModel != null, "Model is null. Please use the upload API to add the model to the Scoring Engine")
+            onComplete(Future {
+              scoringModel.modelMetadata()
+            }) {
               case Success(metadata) => complete(JsObject("model_details" -> metadata.toJson,
-                "input" -> new JsArray(model.input.map(input => FieldFormat.write(input)).toList),
-                "output" -> new JsArray(model.output.map(output => FieldFormat.write(output)).toList)).toString)
+                "input" -> new JsArray(scoringModel.input().map(input => FieldFormat.write(input)).toList),
+                "output" -> new JsArray(scoringModel.output().map(output => FieldFormat.write(output)).toList)).toString())
               case Failure(ex) => ctx => {
                 ctx.complete(StatusCodes.InternalServerError, ex.getMessage)
 
               }
             }
           }
         }
+      } ~
+      path("uploadMarFile") {
+        post {
+          var ret: Option[Route] = None
+          entity(as[MultipartFormData]) { formData =>
+            val file = formData.get("file")
+            for (fileBodypart: BodyPart <- file) {
+              val fileEntity: HttpEntity = fileBodypart.entity
+              val status = installModel(fileEntity.data.toByteArray)
+              if(status == "OK") {
+                ret = Some(complete("File was successfully uploaded and model created. \n"))
+              }
+              else {
+                ret = Some (complete("Model already installed in SE. App does not allow uploading another model. \n"))
```

Comment:
this is only if the Model is already installed in SE. In that case the user gets this message that "Model already installed in SE. App does not allow uploading another model."
If you see a couple of lines below...then if anything else goes wrong...then the user gets internal server error